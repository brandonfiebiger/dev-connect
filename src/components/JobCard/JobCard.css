.job-card {
  background-color: #2e2424;
  border-radius: 25px;
  width: 700px; }
