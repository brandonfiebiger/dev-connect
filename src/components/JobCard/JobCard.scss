.job-card {
  background-color: #2e2424;
  opacity: 0.9;
}
