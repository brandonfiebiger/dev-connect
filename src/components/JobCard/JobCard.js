import React from 'react';
import './JobCard.css';

export const JobCard = (props) => {


  // const jobTypeTitle = () => {
  //   const jobType = props.jobTypes.filter(type => type.id === props.jobTitleId);
  //   console.log(jobType[0])
  //   jobTitle = jobType.job_title
  // }
  // jobTypeTitle();

  return (
    <li>
      {props.description}
    </li>
  )
}