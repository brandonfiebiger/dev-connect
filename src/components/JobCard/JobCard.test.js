import React from 'react';
import { shallow, mount } from 'enzyme';
import { JobCard, mapDispatchToProps, mapStateToProps } from './JobCard';

describe('JobCard', () => {
  let wrapper;
  let mockCompany = 'Gusto';
  let mockLocation = 'Denver, CO';
  let mockType = 'UX Designer';
  let mockDescription = 'We are looking for a talented developer';
  let mockUpdateJob;

  beforeEach(() => {
    wrapper = shallow(
      <JobCard
        description={mockDescription}
        company={mockCompany}
        location={mockLocation}
        jobType={mockType}
        saveJob={jest.fn()}
        removeJob={jest.fn()}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it.skip('should call updateJobSaveStatus when button is clicked', () => {
    mockUpdateJob = jest.fn();

    wrapper = mount(
      <JobCard
        description={mockDescription}
        company={mockCompany}
        location={mockLocation}
        jobType={mockType}
        saveJob={jest.fn()}
        removeJob={jest.fn()}
        updateJobSaveStatus={mockUpdateJob}
      />
    );
    wrapper.find('.update-status-button').simulate('click');
    expect(mockUpdateJob).toHaveBeenCalled();
  });

  it('should call saveJob when the savedJobs array does not have its id', () => {
    let mockId = 52;

    wrapper = mount(
      <JobCard
        description={mockDescription}
        company={mockCompany}
        location={mockLocation}
        jobType={mockType}
        saveJob={jest.fn()}
        removeJob={jest.fn()}
        id={mockId}
        updateJobSaveStatus={mockUpdateJob}
        savedJobs={[]}
      />
    );

    wrapper.instance().handleJobStatus();
    expect(wrapper.props().saveJob).toHaveBeenCalledWith(mockId);
  });

  it('should call removeJob when the savedJobs array does have its id', () => {
    let mockId = 52;

    wrapper = mount(
      <JobCard
        description={mockDescription}
        company={mockCompany}
        location={mockLocation}
        jobType={mockType}
        saveJob={jest.fn()}
        removeJob={jest.fn()}
        id={mockId}
        updateJobSaveStatus={mockUpdateJob}
        savedJobs={[52]}
      />
    );

    wrapper.instance().handleJobStatus();
    expect(wrapper.props().removeJob).toHaveBeenCalledWith(mockId);
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch when using saveJob from mapDispatchToProps', () => {
      let mockId = 52;

      wrapper = mount(
        <JobCard
          description={mockDescription}
          company={mockCompany}
          location={mockLocation}
          jobType={mockType}
          saveJob={jest.fn()}
          removeJob={jest.fn()}
          id={mockId}
          updateJobSaveStatus={mockUpdateJob}
          savedJobs={[]}
        />
      );
      const mockDispatch = jest.fn();

      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.saveJob(mockId);
      expect(mockDispatch).toHaveBeenCalled();
    });

    it('should call dispatch when using removeJob from mapDispatchToProps', () => {
      let mockId = 52;

      wrapper = mount(
        <JobCard
          description={mockDescription}
          company={mockCompany}
          location={mockLocation}
          jobType={mockType}
          saveJob={jest.fn()}
          removeJob={jest.fn()}
          id={mockId}
          updateJobSaveStatus={mockUpdateJob}
          savedJobs={[52]}
        />
      );
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.removeJob(mockId);
      expect(mockDispatch).toHaveBeenCalled();
    });
  });
});
