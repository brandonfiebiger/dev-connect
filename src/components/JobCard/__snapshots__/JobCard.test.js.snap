// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobCard should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <JobCard
    company="Gusto"
    description="We are looking for a talented developer"
    jobType="UX Designer"
    location="Denver, CO"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Gusto
        </h3>,
        <h4>
          UX Designer
        </h4>,
        <p>
          Denver, Co
        </p>,
        <p>
          We are looking for a talented developer
        </p>,
        <button
          onClick={[Function]}
        >
          Save Job
        </button>,
      ],
      "className": "job-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Gusto",
        },
        "ref": null,
        "rendered": "Gusto",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "UX Designer",
        },
        "ref": null,
        "rendered": "UX Designer",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Denver, Co",
        },
        "ref": null,
        "rendered": "Denver, Co",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "We are looking for a talented developer",
        },
        "ref": null,
        "rendered": "We are looking for a talented developer",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Save Job",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Save Job",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Gusto
          </h3>,
          <h4>
            UX Designer
          </h4>,
          <p>
            Denver, Co
          </p>,
          <p>
            We are looking for a talented developer
          </p>,
          <button
            onClick={[Function]}
          >
            Save Job
          </button>,
        ],
        "className": "job-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Gusto",
          },
          "ref": null,
          "rendered": "Gusto",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "UX Designer",
          },
          "ref": null,
          "rendered": "UX Designer",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Denver, Co",
          },
          "ref": null,
          "rendered": "Denver, Co",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "We are looking for a talented developer",
          },
          "ref": null,
          "rendered": "We are looking for a talented developer",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Save Job",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Save Job",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
